example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidLoginSessionFlow.kt

```diff
@@ -0,0 +1,277 @@
+                body.add("grant_type", "authorization_code")
+                body.add("code", code)
+                body.add("redirect_uri", redirectUri)
+                body.add("client_id", clientId)
+                body.add("client_secret", clientSecret)
+                body.add("code_verifier", getOauth2CodeVerifier(session))
+
+                // Exchange code for access token and refresh token
+                val tokenResponse = webClient.post()
+                    .uri(URIBuilder(truidTokenEndpoint).build())
+                    .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .body(fromFormData(body))
+                    .retrieve()
+                    .awaitBody<TokenResponse>()
+
+                // Get and print user email from Truid
+                val getPresentationUri = URIBuilder(truidPresentationEndpoint)
+                    .addParameter("claims", "truid.app/claim/email/v1")
+                    .build()
+
+                val presentation = webClient
+                    .get()
+                    .uri(getPresentationUri)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .header(HttpHeaders.AUTHORIZATION, "Bearer ${tokenResponse.accessToken}")
+                    .retrieve()
+                    .awaitBody<PresentationResponse>()
+
+                println(presentation)
+
+                // Persist token, so it can be accessed via GET "/truid/v1/presentation"
+                // See getAccessToken for an example of refreshing access token
+                persist(tokenResponse)
+            } catch (e: WebClientResponseException.Forbidden) {
+                throw Forbidden("access_denied", e.message)
+            }
+        }
+
+        if (exchange.request.headers.accept.contains(MediaType.TEXT_HTML)) {
+            // Redirect to success page in the webapp flow
+            exchange.response.headers.location = webSuccess
+            exchange.response.statusCode = HttpStatus.FOUND
+            return null
+        } else {
+            // Return a 200 response in case of an AJAX request
+            exchange.response.statusCode = HttpStatus.OK
+            return null
+        }
+    }
+
+    @GetMapping(
+        path = ["/truid/v1/login-presentation"],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    suspend fun getPresentation(): PresentationResponse {
+        val accessToken = refreshToken()
+
+        val getPresentationUri = URIBuilder(truidPresentationEndpoint)
+            .addParameter("claims", "truid.app/claim/email/v1")
+            .build()
+
+        return webClient
+            .get()
+            .uri(getPresentationUri)
+            .accept(MediaType.APPLICATION_JSON)
+            .header(HttpHeaders.AUTHORIZATION, "Bearer $accessToken")
+            .retrieve()
+            .awaitBody()
+    }
+
+    private suspend fun refreshToken(): String {
```

Comment:
The example should perhaps check the expiry time of the access token, an only refresh if it has expired?